ooni/oonicli.py

```diff
@@ -315,10 +320,10 @@ def post_director_start(_):
 
             net_test_loader.annotations = global_options['annotations']
 
-            director.startNetTest(net_test_loader,
-                                    global_options['reportfile'],
-                                    collector)
-        return director.allTestsDone
+            yield director.startNetTest(net_test_loader,
+                                        global_options['reportfile'],
+                                        collector,
+                                        global_options['no-yamloo'])
```

Comment:
Responding to myself: basically ooni has two schedulers, one for tests and one for reports. A test is bound to lots of measurements when you run a deck. Hence, opening all the reports in parallel could lead to timeouts. This changes how ooni works such that only one test at a time is started (but still measurements in the context of such test go in parallel).